vm-device/src/lib.rs

```diff
@@ -10,6 +10,7 @@ use serde::{Deserialize, Serialize};
 mod bus;
 pub mod dma_mapping;
 pub mod interrupt;
+pub mod serial_buffer;
```

Comment:
Urgh, i'm not sure this is a good idea. This really overloads the meaning of the crate. I know it's only a short bit of code but it would be better it it's own crate.